groupby.py

```diff
@@ -1,29 +1,116 @@
-def render(table, params):
-    # obtain parameters
-    groupby = params['groupby']
-    operation = params['operation']
-    targetcolumn = params['targetcolumn']
-
-    # while the user has not provided all the parameters, returns full table.
-    if groupby == '' or (targetcolumn == '' and operation != 0):  # process without target column if counting
-        return table
-    else:
-        # if target column is not a numeric type, tries to convert it (before any aggregation)
-        if targetcolumn != '' and (table[targetcolumn].dtype != np.float64 and table[targetcolumn].dtype != np.int64):
-            table[targetcolumn] = table[targetcolumn].str.replace(',', '')
-            table[targetcolumn] = table[targetcolumn].astype(float)
-
-        if operation == 0: # count
-            newtab = table.groupby([groupby])[[groupby]].count()
-            newtab.columns = ['count'] # otherwise index name and count column name are the same, error on reset_index below
-        elif operation == 1: # average
-            newtab = table.groupby([groupby])[[targetcolumn]].mean()
-        elif operation == 2: # sum
-            newtab = table.groupby([groupby])[[targetcolumn]].sum()
-        elif operation == 3: # min
-            newtab = table.groupby([groupby])[[targetcolumn]].min()
-        elif operation == 4: # max
-            newtab = table.groupby([groupby])[[targetcolumn]].max()
-        newtab.reset_index(level=0, inplace=True)
-
-        return newtab
+import numpy as np
+import pandas as pd
+from io import StringIO
+import re
+import pdb
+
+
+class Importable:
+    @staticmethod
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def render(table, params):
+        param_obj = {}
+        for k, v in params.items():
+            name_parts = k.split('|')
+            if len(name_parts) > 1:
+                try:
+                    param_obj[name_parts[1]]
+                except KeyError:
+                    param_obj[name_parts[1]] = {}
+
+                try:
+                    param_obj[name_parts[1]][name_parts[2]]
+                except:
+                    param_obj[name_parts[1]][name_parts[2]] = {}
+
+                param_obj[name_parts[1]][name_parts[2]][name_parts[0]] = v
+
+        for k, v in param_obj.items():
+            param_obj[k] = [s[1] for s in sorted(v.items(), key=lambda x: x[0])]
+
+        # obtain parameters
+        # For every paramater in group 'groupby', return the 'groupby' property if it's not blank,
+        # property 'active' is set to true, or if there is no property called 'active'.
```

Comment:
What is a "property" here? I guess you're envisioning a future where we have parameter arrays/groups. Rather than having an active flag on each parameter, why not just exclude the parameters which are not in use from the param_obj array?